### PR TITLE
observerward 2022.3.15 (new formula)

### PR DESCRIPTION
**Pull request by Rvn0xsy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/observerward.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match "ObserverWard", shell_output("#{bin}/observer_ward")
```
Comment (carlocab) [inline Formula/observerward.rb:15]: Does this test any functionality, or does this just dump help text?

Comment (Rvn0xsy) [reply Formula/observerward.rb:15]: > 

just dump help text

Comment (Rvn0xsy) [reply Formula/observerward.rb:15]: > Does this test any functionality, or does this just dump help text?

just dump help text

Comment (carlocab) [reply Formula/observerward.rb:15]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Any of the examples from https://github.com/0x727/ObserverWard/#使用方法 (other than `-h`) should be suitable for this.

Comment (carlocab) [reply Formula/observerward.rb:15]: For example, you could [start up a server](https://github.com/0x727/ObserverWard/#开启api服务) like in this example, query it using `curl` and then parse the response. There are many examples of tests like this in this repo. (`grep`ping for `fork do` would be a good start.)

Comment (Rvn0xsy) [reply Formula/observerward.rb:15]: > Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.
> 
> > We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.
> 
> Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula
> 
> Any of the examples from [https://github.com/0x727/ObserverWard/#使用方法](https://github.com/0x727/ObserverWard/#%E4%BD%BF%E7%94%A8%E6%96%B9%E6%B3%95) (other than `-h`) should be suitable for this.

Thanks,I'll update this test.

File: Formula/observerward.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "#{bin}/observer_ward", "-u"
```
Comment (carlocab) [inline Formula/observerward.rb:15]:
```suggestion
    system bin/"observer_ward", "-u"
```
You don't need interpolation here.

**Review by carlocab — COMMENTED** [review]
This is much better. Just one small suggestion.

File: Formula/observerward.rb (hunk 2 above)
Comment (Rvn0xsy) [reply Formula/observerward.rb]: I have made modifications, please check

**Review by carlocab — APPROVED** [review]
Thanks!

For future reference, please put the name of the formula first in your commit message, as mentioned in the commit style guide, like so:
```
observerward 2022.3.15 (new formula)
```

**Pull request closed without merging**